week2/homework-week2/Program.cs

```diff
@@ -0,0 +1,97 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+
+// 1. Calculator
+app.MapGet("/calculate", (string number1, string number2, string operation) => {
+    var number1Parse = int.TryParse(number1, out int value1);
+    var number2Parse = int.TryParse(number2, out int value2);
+  
+    if(!number1Parse || !number2Parse){
+        return Results.Ok(new { Error = $"Unable to parse input '{number1}', '{number2}'!" });  
+    }
+    var result = 0;
+    switch(operation){
+        case "add":
+            result = value1 + value2;
+            break;
+         case "substract":
+            result = value1 - value2;
+            break;
+             case "multiply":
+            result = value1 * value2;
+            break;
+        default:
+          return Results.Ok(new { Error = $"System not supported - please try again"});  
+    }
+    return Results.Ok(new String($"the result is {result}"));
+});
+
+// 2. Method example
+app.MapGet("/input", (string input) =>
+{
+     var inputParse = int.TryParse(input, out int value);
+     if(inputParse){
+            System.Console.WriteLine($"The sum is  {AddNumbers(value)}");
+        } else{
+            System.Console.WriteLine($"The total uppercase is {CountCapitalLetters(input)}");
+        }
+    
+    int AddNumbers(int value){
+        var sum = 0;
+        for (int i = value; i > 0; i = i / 10){
+           sum = sum + i % 10; 
+        }return sum;
+    }
+
+    int CountCapitalLetters(string input){
+        var count = 0;
+        for (int i = 0; i < input.Length; i++){
+            char ch = input[i];
+            if(ch >= 'A' && ch <= 'Z')

```

Comment:
You can also use the inbuild method called something like IsLower that is provided on `char` type to check the casing 😊